prompt-service/src/unstract/prompt_service/main.py

```diff
@@ -592,6 +594,18 @@ def prompt_processor() -> Any:
                         f"No eval plugin found to evaluate prompt: {output[PSKeys.NAME]}"  # noqa: E501
                     )
         finally:
+            challenge_metrics = (
+                {f"{challenge_llm.get_usage_reason()}_llm": challenge_llm.get_metrics()}
+                if enable_challenge
+                else {}
+            )
+            metrics.setdefault(prompt_name, {}).update(
+                {
+                    "context_retrieval": index.get_metrics(),
```

Comment:
@Deepak-Kesavan I think such usage reasons of `context_retrieval` should be moved into the corresponding SDK method that returns such a value. If I'm correct only `index.query_index()` corresponds to this. We might also need to have provision to pass another optional reason key that such callers can pass to indicate the use of an SDK method

Reply:
@chandrasekharan-zipstack We can handle it iteratively when we display metrics in a more granular format.